docs/adr/0015-cluster-resources-amendment.md

```diff
@@ -0,0 +1,35 @@
+# Declarative Definitions for Cluster Scoped Resources
+
+- Status: accepted
+- Deciders: larsks, HumairAK, tumido
+
+Technical Story: [issue-1][1]
+
+## Context and Problem Statement
+
+In [ADR-0015][2] we laid out a directory structure of organizing cluster scoped resources. For the most part this has worked out relatively well. However, one thing that was not factored into this structure were the possibilities of name conflicts. Consider the following scenario:
+
+Suppose we have:
+1) A resource of kind Project named example-project with an apiversion: project.openshift.io/v1
+2) A resource of kind Project named example-project with an apiversion: config.openshift.io/v1
+
+Note that their API group differs, but everything else is the same. Both of these resources can exist simultaneously on a live cluster. However, these cannot co-exist in our github repo at `operate-first/apps/cluster-scope`, if we are to use the structuring as defined in [ADR-0015][2]. This is because, using the current structure, they would both need to be stored under `operate-first/apps/cluster-scope/base/projects/example-project/project.yaml`.
+
+
+## Considered Options
+
+The general consensus has been to keep the current structure and amend it to factor in the API Group. Some suggestions are:
+
+1. Use the following schema: `cluster-scope/base/<API_GROUP>-<KIND_PLURAL>/<METADATA_NAME>/<KIND_SINGULAR>.yaml`
+  This would factor in the API group as part of the name, and would not require additional nesting of directories
+2. Use the following schema: `cluster-scoped-resources/base/<API_GROUP>/<KIND_PLURAL>/<NAME>.yaml`
+  This would require additional nesting of directories, but this method has been observed in other tooling (i.e. how must-gather does their organization)
```

Comment:
Actually, is this going to work?  If we have a `kustomization.yaml` that looks like:

```
resources:
- ../../../base/core/namespaces/democratic-csi.yaml
```

Then `kustomize` will complain:

```
Error: accumulating resources: accumulation err='accumulating resources from '../../../base/core/namespaces/democratic-csi.yaml': security; file '/path/to/repo/cluster-scope/base/core/namespaces/democratic-csi.yaml' is not in or below '/path/to/repo/cluster-scope/overlay/oai-demo'': got file 'democratic-csi.yaml', but '/path/to/repo/cluster-scope/base/core/namespaces/democratic-csi.yaml' must be a directory to be a root
```

That is, `kustomize` will only include directories, not files, that
are outside the current directory.

Reply:
By "blank" you mean a kustomization file that only includes a single file as a resource?

Reply:
This also means we need to use `cluster-scoped-resources/base/<API_GROUP>/<KIND_PLURAL>/<NAME>/<KIND>.yaml`, since the kustomization file will need to live as a sibling to the manifest.

Reply:
So we would end up with:

```
.
├── config.openshift.io
│   ├── oauths
│   │   └── cluster
│   │       ├── kustomization.yaml
│   │       └── oauth.yaml
│   └── projects
│       └── cluster
│           ├── kustomization.yaml
│           └── project.yaml
[...]
```

And for example `config.openshift.io/oauths/cluster/kustomization.yaml` would look like:

```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
- oauth.yaml
```

Reply:
updated the adr to reflect the suggestion to include kustomization.yaml per resource

Reply:
@larsks yes. By blank I've meant the same. It means the current scheme for each resource (a resource file and a kustomization including this single resource) would stay unchanged.

Reply:
Another reason to keep the `kustomization.yaml` files is that we're using them in many cases to set resource namespaces (see, e.g., `base/namespaces/advanced-cluster-management/kustomization.yaml`). If we drop the `kustomization.yaml`, that metadata gets lost.